activerecord/lib/active_record/secure_password.rb

```diff
@@ -33,6 +33,9 @@ module ClassMethods
       #   User.authenticate_by(email: "jdoe@example.com", password: "wrong")       # => nil (in 373.9ms)
       #   User.authenticate_by(email: "wrong@example.com", password: "abc123")     # => nil (in 373.6ms)
       #
+      #   User.authenticate_by(email: "jdoe@example.com", password: nil) # => nil
+      #   User.authenticate_by(email: "jdoe@example.com", password: "")  # => nil
```

Comment:
To clarify this further, what if we add a note similar to the "(in ###.#ms)" notes?  Something like:

```suggestion
      #   User.authenticate_by(email: "jdoe@example.com", password: nil) # => nil (no records fetched)
      #   User.authenticate_by(email: "jdoe@example.com", password: "")  # => nil (no records fetched)
```

Or 

```suggestion
      #   User.authenticate_by(email: "jdoe@example.com", password: nil) # => nil (no queries executed)
      #   User.authenticate_by(email: "jdoe@example.com", password: "")  # => nil (no queries executed)
```

Reply:
thank you reviewing @jonathanhefner I'll update.